client/clientfactory.go

```diff
@@ -156,8 +157,13 @@ func (cf *rpcClientFactory) NewHistoryClientWithTimeout(timeout time.Duration) (
 		return cf.newHistoryThriftClient(clientKey)
 	}
 
+	defaultMessageSize := cf.rpcFactory.GetGRPCMaxSizeInBytes()
+	if cf.enableGRPCOutbound && defaultMessageSize == 0 {
+		defaultMessageSize = common.DefaultGRPCSizeLimit
```

Comment:
I guess we don't have to make this specific to grpc?

Reply:
+1
I also think the entire logic with defaults and retrieval from dynamic could be within rpcFactory iself. Here we could just use the final resolved size `rpcFactory.GetMaxMessageSize()`

Reply:
Do you mean we should also limit the size with yarpc?

Reply:
I meant that this `if` statement together with `common.DefaultGRPCSizeLimit` could go inside `rpc` package. It is not strictly required, just some code organization.